android/app/src/main/java/org/jitsi/meet/AudioModeModule.java

```diff
@@ -0,0 +1,440 @@
+package org.jitsi.meet.audiomode;
+
+import android.annotation.TargetApi;
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothDevice;
+import android.bluetooth.BluetoothHeadset;
+import android.bluetooth.BluetoothProfile;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.media.AudioDeviceInfo;
+import android.media.AudioManager;
+import android.os.Build;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Module implementing a simple API to select the appropriate audio device for a conference call.
+ *
+ * Audio calls should use <tt>AudioModeModule.AUDIO_CALL</tt>, which uses the builtin earpiece,
+ * wired headset or bluetooth headset. The builtin earpiece is the default audio device.
+ *
+ * Video calls should should use <tt>AudioModeModule.VIDEO_CALL</tt>, which uses the builtin
+ * speaker, earpiece, wired headset or bluetooth headset. The builtin speaker is the default
+ * audio device.
+ *
+ * Before a call has started and after it has ended the <tt>AudioModeModule.DEFAULT</tt> mode
+ * should be used.
+ */
+public class AudioModeModule extends ReactContextBaseJavaModule {
+    /**
+     * Constants representing the audio mode.
+     */
+    private static final int DEFAULT    = 0;
+    private static final int AUDIO_CALL = 1;
+    private static final int VIDEO_CALL = 2;
+
+    /**
+     * React Native module name.
+     */
+    private static final String MODULE_NAME = "AudioMode";
+
+    /**
+     * Tag used when logging messages.
+     */
+    private static final String TAG = MODULE_NAME;
+
+    /**
+     * Audio mode currently in use.
+     */
+    private int currentMode = -1;
```

Comment:
I don't particularly like the `current` naming. It's a field/state of the instance, so the current is implied.